### PR TITLE
1173 ucn edit modal

### PR DESCRIPTION
**Pull request by momimomo**

### CODE REVIEW

File: frontend/src/components/eois/modals/editUcn/editUcnModal.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+import { submit } from 'redux-form';
+import Grid from 'material-ui/Grid';
+import ControlledModal from '../../../common/modals/controlledModal';
+import EditUcnForm from './editUcnForm';
+import { updateUcn } from '../../../../reducers/newCfei';
+import { errorToBeAdded } from '../../../../reducers/errorReducer';
+import { selectCountriesWithOptionalLocations } from '../../../../store';
+
+const messages = {
+  title: 'Edit direct selection/retention',
```
Comment (kenzo23312) [inline frontend/src/components/eois/modals/editUcn/editUcnModal.js]: edit unsoli...

File: frontend/src/components/eois/modals/editUcn/editUcnModal.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+  save: 'Save',
+  error: 'Unable to update direct selection/retention',
```
Comment (kenzo23312) [inline frontend/src/components/eois/modals/editUcn/editUcnModal.js]: unsoli....

File: frontend/src/reducers/cfeiDetails.js [hunk 3]
```diff
@@ -116,9 +116,10 @@ const saveUCN = (state, action) => {
     partner_name: ucn.partner.legal_name,
     display_type: ucn.partner.display_type,
     title: R.path(['proposal_of_eoi_details', 'title'], ucn),
-    locations: normalizeLocations(ucn.locations_proposal_of_eoi),
+    locations: normalizeEditLocations(ucn.locations_proposal_of_eoi),
```
Comment (kenzo23312) [inline frontend/src/reducers/cfeiDetails.js]: we use `locations` field please add extra locationEdit or similar to use in initialValues

File: frontend/src/reducers/cfeiDetails.js [hunk 4]
```diff
@@ -132,6 +133,11 @@ export function selectCfeiDetail(state, id) {
   return cfei;
 }
 
+export function selectUcnDetail(state, id) {
```
Comment (kenzo23312) [inline frontend/src/reducers/cfeiDetails.js]: remove

File: frontend/src/reducers/newCfei.js [hunk 5]
```diff
@@ -136,6 +136,27 @@ export const updateCfei = (body, id) => (dispatch, getState) => patchCfei(body,
     });
   });
 
+export const updateUcn = (body, id) => (dispatch, getState) => patchUcn(body, id)
+  .then((cfei) => {
+    dispatch(loadCfeiDetailSuccess(cfei));
+    if (cfei.direct_selected_partners) {
```
Comment (kenzo23312) [inline frontend/src/reducers/newCfei.js]: save updated UCN to state

File: frontend/src/reducers/cfeiDetails.js (hunk 3 above)
Comment (momimomo) [reply frontend/src/reducers/cfeiDetails.js]: ? It's used to create request with different field names, however in backend it's stored as such - it's displaying properly.

**Pull request merged**